numpy/core/src/multiarray/array_coercion.c

```diff
@@ -542,9 +543,9 @@ PyArray_Pack(PyArray_Descr *descr, char *item, PyObject *value)
     int res = cast_raw_scalar_item(tmp_descr, data, descr, item);
 
     if (PyDataType_REFCHK(tmp_descr)) {
-        /* We could probably use move-references above */
-        PyArray_Item_XDECREF(data, tmp_descr);
+        PyArray_ClearBuffer(tmp_descr, data, 0, 1, 1);
```

Comment:
```suggestion
        if (PyArray_ClearBuffer(tmp_descr, data, 0, 1, 1) < 0) {
            res = -1;
        }
```
Just realized this can fail (not that it ever *should* up to the point that maybe this should be a critical error), most other places are in things like dealloc where errors are not possible.